hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -125,7 +127,9 @@
   private List<CompletableFuture<DataStreamReply>> futures = new ArrayList<>();
   private final long syncSize = 0; // TODO: disk sync is disabled for now
   private long syncPosition = 0;
-
+  private StreamBuffer currentBuffer;
+  private int currentBufferRemaining;
+  private XceiverClientMetrics metrics;
```

Comment:
Do we really need to track currentBufferRemaining? It can always be deduced from streamBuffer.

Reply:
Done.